src/main/java/com/amazon/opendistroforelasticsearch/security/securityconf/impl/WhitelistingSettings.java

```diff
@@ -0,0 +1,112 @@
+package com.amazon.opendistroforelasticsearch.security.securityconf.impl;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestStatus;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class WhitelistingSettings {
+    private boolean enabled;
+    private Map<String, List<HttpRequestMethods>> requests;
+
+    /**
+     * Used to parse the yml files, do not remove.
+     */
+    public WhitelistingSettings() {
+        enabled = false;
+        requests = Collections.emptyMap();
+    }
+
+    public WhitelistingSettings(WhitelistingSettings whitelistingSettings) {
+        this.enabled = whitelistingSettings.getEnabled();
+        this.requests = whitelistingSettings.getRequests();
+    }
+
+    public boolean getEnabled() {
+        return this.enabled;
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    public Map<String, List<HttpRequestMethods>> getRequests() {
+        return this.requests == null ? Collections.emptyMap(): this.requests;
```

Comment:
nit: consider moving this null check to setter as getter is more likely accessed